### PR TITLE
Adds new Magic Weather sample app

### PR DESCRIPTION
**Pull request by codykerns**
This PR adds the new 'MagicWeather' sample app to a new /Examples folder (for platform consistency).

Also leaves the existing example folder in place.

### CODE REVIEW

**Review by aboedo — COMMENTED** [review]
Looks great! Leaving a couple of questions: 

- do we actually need to have a `lib/main.dart`? I'm not sure if running straight from terminal works without it
- would having it in `Examples` actually be more confusing for developers? we now have to folders: `example` and `Examples`

**Comment by vegaro** [comment]
Sorry I missed this. Looking now

File: Examples/MagicWeather/lib/src/components/native_dialog.dart [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    if (!Platform.isIOS) {
+      return AlertDialog(
+        title: new Text(
+          title,
+        ),
+        content: new Text(
+          this.content,
+        ),
+        actions: <Widget>[
+          new FlatButton(
```
Comment (vegaro) [inline Examples/MagicWeather/lib/src/components/native_dialog.dart:23]: There's a warning that says `FlatButton` is deprecated and `TextButton` should be used

File: Examples/MagicWeather/android/app/src/debug/AndroidManifest.xml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+    package="m4trixdevlimited.com.flutter_magic_weather_app">
```
Comment (vegaro) [inline Examples/MagicWeather/android/app/src/debug/AndroidManifest.xml:2]: should we update this package name? in other examples we have `com.revenuecat.sample` if I am not wrong

File: Examples/MagicWeather/android/app/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -0,0 +1,51 @@
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+    package="m4trixdevlimited.com.flutter_magic_weather_app">
```
Comment (vegaro) [inline Examples/MagicWeather/android/app/src/main/AndroidManifest.xml:2]: same question here about the package name

**Review by vegaro — COMMENTED** [review]
Besides the package name it looks good to me. I am not sure why the lint is complaining. I am going to look into it. It only complains when running `flutter analyze` in the root which is strange

**Comment by vegaro** [comment]
Just fixed the lint job, it was missing the `pub get` in the example folder. Apparently running `pub get` in the root only does pub get in the package and in the current example, but not in Examples/MagicWeather. Anyhow, it shows the only real issue now

**Review by aboedo — APPROVED** [review]

**Pull request merged**